hugr/src/algorithm/nest_cfgs.rs

```diff
@@ -855,39 +860,12 @@ pub(crate) mod test {
         Ok(merge)
     }
 
-    // Returns loop tail - caller must link header to tail, and provide 0th successor of tail
```

Comment:
I think it was these awkward interfaces "returns x, caller must do y", which are fairly essential for building "bits of CFG", that lead me to drop them; it is hard to tell that the callsites are correct without jumping around a lot

Reply:
I think I agree, inlining these is a better move 👍